dix/cursor.c

```diff
@@ -200,8 +200,8 @@ RealizeCursorAllScreens(CursorPtr pCurs)
                             walkScreen->UnrealizeCursor(pDevIt, walkScreen, pCurs);
                         pDevIt = pDevIt->next;
                     }
-                    while (--nscr >= 0) {
-                        walkScreen = screenInfo.screens[nscr];
+                    while (--walkScreenIdx>= 0) {
```

Comment:
If `walkScreenIdx` is 0 here, then the first decrement will give us a big positive value (because `walkScreenIdx` is unsigned).